client/asset/eth/eth.go

```diff
@@ -384,12 +385,12 @@ func (eth *ExchangeWallet) balanceImpl() (*asset.Balance, error) {
 // result is cached and reused in subsequent requests.
 //
 // This function will return an error if the wallet has a non zero balance.
-func (eth *ExchangeWallet) getInitGas() (uint64, error) {
-	eth.cachedInitGasMtx.Lock()
-	defer eth.cachedInitGasMtx.Unlock()
+func (eth *ExchangeWallet) getInitGas(numSwaps int) (uint64, error) {
+	eth.initGasCacheMtx.Lock()
+	defer eth.initGasCacheMtx.Unlock()
 
-	if eth.cachedInitGas > 0 {
-		return eth.cachedInitGas, nil
+	if cachedGas, ok := eth.initGasCache[numSwaps]; ok {
+		return cachedGas, nil
 	}
```

Comment:
I'm not crazy about this solution. Have we decided against using constants derived from your previous results?

```
Gas used for batch initiating 1 swaps: 180037. 180037 more than previous 
Gas used for batch initiating 2 swaps: 338287. 158250 more than previous 
Gas used for batch initiating 3 swaps: 496537. 158250 more than previous 
Gas used for batch initiating 4 swaps: 654775. 158238 more than previous 
Gas used for batch initiating 5 swaps: 813025. 158250 more than previous 
Gas used for batch initiating 6 swaps: 971275. 158250 more than previous 
Gas used for batch initiating 7 swaps: 1129513. 158238 more than previous 
Gas used for batch initiating 8 swaps: 1287763. 158250 more than previous 
Gas used for batch initiating 9 swaps: 1446025. 158262 more than previous
```

Reply:
Yeah I think we might as well do that. It'll make everything more simple.

Reply:
How about I do a separate PR for this.

Reply:
Not a priority. I'm already using `InitGas = 181000` in some in-progress work.

Reply:
It's actually updated to `InitGas = 157000` now. I also have `AdditionalInitGas` in there now, but I don't think you should worry about. Generally you should just assume the worst case scenario.

Reply:
I don't follow. What is `AdditionalInitGas` and why would `InitGas` it be 157000? A single lot swap is `180037`, so we do `lots * 180037` to get the required fee reserves. So `SwapSize = SwapSizeBase = InitGas = 181000` (with `inputsSize = 0` in e.g. `calc.RequiredOrderFunds`) makes the math work out everywhere.

Reply:
Ah. Values are outdated. 

```
Gas used for batch initiating 1 swaps: 156485. 156485 more than previous 
Gas used for batch initiating 2 swaps: 291127. 134642 more than previous 
Gas used for batch initiating 3 swaps: 425793. 134666 more than previous
```